database/ffldb/transaction_test.go

```diff
@@ -374,43 +364,160 @@ func TestTransactionCloseErrors(t *testing.T) {
 
 	for _, test := range tests {
 		func() {
-			// Open a test db
-			path, err := ioutil.TempDir("", "TestTransactionCloseErrors")
-			if err != nil {
-				t.Fatalf("TestTransactionCloseErrors: TempDir unexpectedly "+
-					"failed: %s", err)
-			}
-			db, err := Open(path)
+			db, teardownFunc := prepareDatabaseForTest(t, "TestTransactionCommitErrors")
+			defer teardownFunc()
+
+			// Begin a new transaction
+			dbTx, err := db.Begin()
 			if err != nil {
-				t.Fatalf("TestTransactionCloseErrors: Open "+
+				t.Fatalf("TestTransactionCommitErrors: Begin "+
 					"unexpectedly failed: %s", err)
 			}
 			defer func() {
-				err := db.Close()
+				err := dbTx.RollbackUnlessClosed()
 				if err != nil {
-					t.Fatalf("TestTransactionCloseErrors: Close "+
+					t.Fatalf("TestTransactionCommitErrors: RollbackUnlessClosed "+
 						"unexpectedly failed: %s", err)
 				}
 			}()
 
+			// Commit the transaction
+			err = dbTx.Commit()
+			if err != nil {
+				t.Fatalf("TestTransactionCommitErrors: Commit "+
+					"unexpectedly failed: %s", err)
+			}
+
+			expectedErrContainsString := "closed transaction"
+
+			// Make sure that the test function returns a "closed transaction" error
+			err = test.function(dbTx)
+			if test.shouldReturnError {
+				if err == nil {
+					t.Fatalf("TestTransactionCommitErrors: %s "+
+						"unexpectedly succeeded", test.name)
+				}
+				if !strings.Contains(err.Error(), expectedErrContainsString) {
+					t.Fatalf("TestTransactionCommitErrors: %s "+
+						"returned wrong error. Want: %s, got: %s",
+						test.name, expectedErrContainsString, err)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("TestTransactionCommitErrors: %s "+
+						"unexpectedly failed: %s", test.name, err)
+				}
+			}
+		}()
+	}
+}
+
+func TestTransactionRollbackErrors(t *testing.T) {
```

Comment:
Can't we somehow combine TestTransactionCommitErrors and TestTransactionRollbackErrors?
They both test for basically the same thing - that after tx close you can't do anything? Am I right?